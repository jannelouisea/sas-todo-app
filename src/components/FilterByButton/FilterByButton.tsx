import Button from '@mui/joy/Button';
import PropTypes from 'prop-types';
import {
    JoyUIColor,
    JoyUIVariant,
    JoyUISize
} from 'src/static/enums'

type Props = {
    label: string,
    color: JoyUIColor,
    filterEnabled: boolean,
    startIcon: React.ReactNode,
    onClick: React.MouseEventHandler<HTMLButtonElement>,
    className: string,
}

function FilterByButton({ label, color, filterEnabled, startIcon, onClick, className }: Props) {
    return (
        <div className={className}>
            <Button
                className={className}
                startDecorator={startIcon}
                size={JoyUISize.Small}
                sx={{ borderRadius: '4rem' }}
                variant={filterEnabled ? JoyUIVariant.Solid : JoyUIVariant.Soft}
                color={filterEnabled ? color : JoyUIColor.Neutral}
                onClick={onClick}
                aria-label='Filter by button'
            >
                {label}
            </Button>
        </div>
    );
}

FilterByButton.propTypes = {
    label: PropTypes.string,
    color: PropTypes.oneOf([JoyUIColor.Primary, JoyUIColor.Success, JoyUIColor.Warning]),
    filterEnabled: PropTypes.bool,
    startIcon: PropTypes.object,
    onClick: PropTypes.object,
    className: PropTypes.string,
}

FilterByButton.defaultProps = {
    filterEnabled: true,
    className: '',
    startIcon: <></>,
}

export default FilterByButton;