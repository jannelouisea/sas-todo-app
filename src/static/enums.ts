export enum Sort {
    Incmpl1st = 'INCMPL_1ST',
    Cmpl1st = 'CMPL_1ST',
    DateAsc = 'DATE_ASC',
    DateDesc = 'DATE_DESC',
    TextAsc = 'TEXT_ASC',
    TextDesc = 'TEXT_DESC',
}

export enum MUIColor {
    Primary = 'primary',
    Success = 'success',
    Warning = 'warning',
    Neutral = 'neutral',
    Danger = 'danger',
}

export enum MUIIconColor {
    Inherit = 'inherit',
}

export enum MUIIconSize {
    Inherit = 'inherit',
}

export enum MUIVariant {
    Solid = 'solid',
    Soft = 'soft',
    Outlined = 'outlined',
    Plain = 'plain',
}

export enum MUISize {
    Small = 'sm',
    Medium = 'md',
    Large = 'lg',
}

export enum MUIPos {
    Right = 'right',
}

export enum MUIText {
    Label = 'body3'
}

export enum MUIFontSize {
    Small = 'small'
}