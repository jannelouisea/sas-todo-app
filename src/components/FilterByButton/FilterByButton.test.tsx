import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import FilterByButton from './FilterByButton';
import { JoyUIColor } from 'src/static/enums';

describe('FilterByButton', () => {
    const labelTextId = 'Filter by button';

    it('renders correctly with default props', () => {
        const tree = renderer.create(
            <FilterByButton
                label='foo'
                color={JoyUIColor.Primary}
                onClick={() => { }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when filter enabled', () => {
        const tree = renderer.create(
            <FilterByButton
                label='foo'
                color={JoyUIColor.Primary}
                onClick={() => { }}
                filterEnabled={true}
                className='foo'
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when filter disabled', () => {
        const tree = renderer.create(
            <FilterByButton
                label='foo'
                color={JoyUIColor.Primary}
                onClick={() => { }}
                filterEnabled={false}
                className='foo'
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('calls onClick prop function on click', () => {
        const onClickSpy = jest.fn();

        render(
            <FilterByButton
                label='foo'
                color={JoyUIColor.Primary}
                onClick={onClickSpy}
            />
        );

        const button = screen.getByLabelText(labelTextId);
        expect(button).toBeInTheDocument();

        fireEvent.click(button);
        expect(onClickSpy).toHaveBeenCalledTimes(1);
    });

});