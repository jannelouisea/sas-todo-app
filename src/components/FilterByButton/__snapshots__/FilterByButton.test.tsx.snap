// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByButton renders correctly when filter disabled 1`] = `
<div
  className="foo"
>
  <button
    aria-label="Filter by button"
    className="MuiButton-root MuiButton-variantSoft MuiButton-colorNeutral MuiButton-sizeSm foo css-d4z7sw-JoyButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
    />
    foo
  </button>
</div>
`;

exports[`FilterByButton renders correctly when filter enabled 1`] = `
<div
  className="foo"
>
  <button
    aria-label="Filter by button"
    className="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeSm foo css-1d374uw-JoyButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
    />
    foo
  </button>
</div>
`;

exports[`FilterByButton renders correctly with default props 1`] = `
<div
  className=""
>
  <button
    aria-label="Filter by button"
    className="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeSm css-1d374uw-JoyButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="MuiButton-startDecorator css-187ulk2-JoyButton-startDecorator"
    />
    foo
  </button>
</div>
`;
